Stream entryid null will cause error

**Describe the problem**
if the stream is newly created, its first entry and last entry will be null in redis and the exporter seems expected the entry id always has value, so it is causing an error in this case.

**What version of redis_exporter are you running?**
Please run `redis_exporter --version` if you're not sure what version you're running.
[ ] 0.3x.x
[X] 1.x.x


**Running the exporter**
What's the full command you're using to run the exporter? (please remove passwords and other sensitive data)
run in docker with the args : --debug=true --export-client-list=true --include-system-metrics=true 


**Expected behavior**
What metrics are missing? What metrics are wrong? Is something missing that was present in an earlier version? 
Did you upgrade from 0.3x.x to 1.0 and are scraping multiple hosts? [Have a look here ](https://github.com/oliver006/redis_exporter#prometheus-configuration-to-scrape-multiple-redis-hosts) how the configuration changed.


**Screenshots**
If applicable, add screenshots to help explain your problem.


**Additional context**

The app will crash. Here is the logs for the error. 

```
panic: interface conversion: interface () is nil, not []interface ()

goroutine 26 [running]:
github.com/oliver006/redis_exporter/exporter.getEntryId(...)
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/src/github.com/oliver006/redis_exporter/exporter/streams.go:70
github.com/oliver006/redis_exporter/exporter.getstreamInfo((0x9f8e20, 0xc0001270e0), [0xc000339240, 0x1b])
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/src/github.com/oliver006/redis_exporter/exporter/streams.go:55 +0x411
github.com/oliver006/redis_exporter/exporter.(*Exporter).extractStreamMetrics(0xc00012c240, 0x5?, (0x9f8e20, 0xc0001270e0])
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/src/github.com/oliver006/redis_exporter/exporter/streams.go:178 +0x554
github.com/oliver006/redis_exporter/exporter.(*Exporter).scrapeRedisHost(0xc00012c240, Oxc0001003c0%)
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/src/github.com/oliver006/redis_exporter/exporter/exporter.go:645 +0xe9b
github.com/oliver006/redis_exporter/exporter.(*Exporter).Collect(0xc00012c240, 0xc00004bf60?)
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/src/github.com/oliver006/redis_exporter/exporter/exporter.go:471 +0xef
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/pkg/mod/github.com/prometheus/client_golang@v1.16.0/prometheus/registry.go:455 +0x16
eated by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
/bitnami/blacksmith-sandox/redis-exporter-1.52.0/pkg/mod/github.com/prometheus/client_golang@v1.16.0/prometheus/registry.go:547
```

--- a/exporter/exporter_test.go
+++ b/exporter/exporter_test.go
@@ -40,11 +40,6 @@ var (
 	altDBNumStr     = "12"
 	invalidDBNumStr = "16"
 	dbNumStrFull    = fmt.Sprintf("db%s", dbNumStr)
-
-	TestStreamTimestamps = []string{
-		"1638006862416-0",
-		"1638006862417-2",
-	}
 )
 
 const (

--- a/exporter/keys_test.go
+++ b/exporter/keys_test.go
@@ -283,7 +283,7 @@ func createKeyFixtures(t *testing.T, c redis.Conn, fixtures []keyFixture) {
 	for _, f := range fixtures {
 		args := append([]interface{}{f.key}, f.args...)
 		if _, err := c.Do(f.command, args...); err != nil {
-			t.Errorf("Error creating fixture: %#v, %#v", f, err)
+			t.Fatalf("Error creating fixture: %#v, %#v", f, err)
 		}
 	}
 }

--- a/exporter/metrics.go
+++ b/exporter/metrics.go
@@ -83,7 +83,7 @@ func (e *Exporter) registerConstMetric(ch chan<- prometheus.Metric, metric strin
 	description := e.findOrCreateMetricDescription(metric, labelValues)
 	m, err := prometheus.NewConstMetric(description, valType, val, labelValues...)
 	if err != nil {
-		log.Debugf("registerConstMetric( %s , %.2f) err: %s", metric, err, val)
+		log.Debugf("registerConstMetric( %s , %.2f) err: %s", metric, val, err)
 		return
 	}
 

--- a/exporter/streams.go
+++ b/exporter/streams.go
@@ -40,40 +40,40 @@ type streamGroupConsumersInfo struct {
 }
 
 func getStreamInfo(c redis.Conn, key string) (*streamInfo, error) {
-	v, err := redis.Values(doRedisCmd(c, "XINFO", "STREAM", key))
+	values, err := redis.Values(doRedisCmd(c, "XINFO", "STREAM", key))
 	if err != nil {
 		return nil, err
 	}
 
 	// Scan slice to struct
 	var stream streamInfo
-	if err := redis.ScanStruct(v, &stream); err != nil {
+	if err := redis.ScanStruct(values, &stream); err != nil {
 		return nil, err
 	}
 
 	// Extract first and last id from slice
-	stream.FirstEntryId = getEntryId(v, 17)
-	stream.LastEntryId = getEntryId(v, 19)
+	stream.FirstEntryId = getStreamEntryId(values, 17)
+	stream.LastEntryId = getStreamEntryId(values, 19)
 
 	stream.StreamGroupsInfo, err = scanStreamGroups(c, key)
 	if err != nil {
 		return nil, err
 	}
 
-	log.Debugf("stream: %#v", &stream)
+	log.Debugf("getStreamInfo() stream: %#v", &stream)
 	return &stream, nil
 }
 
-func getEntryId(redisValue []interface{}, index int) string {
-	var emptyStreamId = ""
-
-	if len(redisValue) < index || len(redisValue) > index && len(redisValue[index].([]interface{})) < 2 {
-		return emptyStreamId
+func getStreamEntryId(redisValue []interface{}, index int) string {
+	if len(redisValue) < index || redisValue[index] == nil || len(redisValue[index].([]interface{})) < 2 {
+		log.Debugf("Failed to parse StreamEntryId")
+		return ""
 	}
 
 	entryId, ok := redisValue[index].([]interface{})[0].([]byte)
 	if !ok {
-		return emptyStreamId
+		log.Debugf("Failed to parse StreamEntryId")
+		return ""
 	}
 	return string(entryId)
 }
@@ -141,9 +141,17 @@ func scanStreamGroupConsumers(c redis.Conn, stream string, group string) ([]stre
 }
 
 func parseStreamItemId(id string) float64 {
+	if strings.TrimSpace(id) == "" {
+		return 0
+	}
+	frags := strings.Split(id, "-")
+	if len(frags) == 0 {
+		log.Errorf("Couldn't parse StreamItemId: %s", id)
+		return 0
+	}
 	parsedId, err := strconv.ParseFloat(strings.Split(id, "-")[0], 64)
 	if err != nil {
-		log.Errorf("Couldn't parse given stream timestamp: %s", err)
+		log.Errorf("Couldn't parse given StreamItemId: [%s]   err: %s", id, err)
 	}
 	return parsedId
 }

--- a/exporter/streams_test.go
+++ b/exporter/streams_test.go
@@ -18,6 +18,13 @@ type scanStreamFixture struct {
 	consumers  []streamGroupConsumersInfo
 }
 
+var (
+	TestStreamTimestamps = []string{
+		"1638006862416-0",
+		"1638006862417-2",
+	}
+)
+
 func isNotTestTimestamp(returned string) bool {
 	for _, expected := range TestStreamTimestamps {
 		if parseStreamItemId(expected) == parseStreamItemId(returned) {
@@ -27,7 +34,7 @@ func isNotTestTimestamp(returned string) bool {
 	return true
 }
 
-func TestGetStreamInfo(t *testing.T) {
+func TestStreamsGetStreamInfo(t *testing.T) {
 	if os.Getenv("TEST_REDIS_URI") == "" {
 		t.Skipf("TEST_REDIS_URI not set - skipping")
 	}
@@ -67,34 +74,34 @@ func TestGetStreamInfo(t *testing.T) {
 			}
 
 			if info.Length != tst.streamInfo.Length {
-				t.Errorf("Stream length mismatch.\nActual: %#v;\nExpected: %#v\n", info.Length, tst.streamInfo.Length)
+				t.Errorf("Stream length mismatch.\nActual: %#v;\nExpected: %#v", info.Length, tst.streamInfo.Length)
 			}
 			if info.RadixTreeKeys != tst.streamInfo.RadixTreeKeys {
-				t.Errorf("Stream RadixTreeKeys mismatch.\nActual: %#v;\nExpected: %#v\n", info.RadixTreeKeys, tst.streamInfo.RadixTreeKeys)
+				t.Errorf("Stream RadixTreeKeys mismatch.\nActual: %#v;\nExpected: %#v", info.RadixTreeKeys, tst.streamInfo.RadixTreeKeys)
 			}
 			if info.RadixTreeNodes != tst.streamInfo.RadixTreeNodes {
-				t.Errorf("Stream RadixTreeNodes mismatch.\nActual: %#v;\nExpected: %#v\n", info.RadixTreeNodes, tst.streamInfo.RadixTreeNodes)
+				t.Errorf("Stream RadixTreeNodes mismatch.\nActual: %#v;\nExpected: %#v", info.RadixTreeNodes, tst.streamInfo.RadixTreeNodes)
 			}
 			if info.Groups != tst.streamInfo.Groups {
-				t.Errorf("Stream Groups mismatch.\nActual: %#v;\nExpected: %#v\n", info.Groups, tst.streamInfo.Groups)
+				t.Errorf("Stream Groups mismatch.\nActual: %#v;\nExpected: %#v", info.Groups, tst.streamInfo.Groups)
 			}
 			if isNotTestTimestamp(info.LastGeneratedId) {
-				t.Errorf("Stream LastGeneratedId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.LastGeneratedId, TestStreamTimestamps)
+				t.Errorf("Stream LastGeneratedId mismatch.\nActual: %#v;\nExpected any of: %#v", info.LastGeneratedId, TestStreamTimestamps)
 			}
 			if info.FirstEntryId != "" {
-				t.Errorf("Stream FirstEntryId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.FirstEntryId, "")
+				t.Errorf("Stream FirstEntryId mismatch.\nActual: %#v; - Expected empty", info.FirstEntryId)
 			}
 			if info.LastEntryId != "" {
-				t.Errorf("Stream LastEntryId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.LastEntryId, "")
+				t.Errorf("Stream LastEntryId mismatch.\nActual: %#v;\nExpected empty", info.LastEntryId)
 			}
 			if info.MaxDeletedEntryId != "" {
-				t.Errorf("Stream MaxDeletedEntryId mismatch.\nActual: %#v;\nExpected: %#v\n", info.MaxDeletedEntryId, "")
+				t.Errorf("Stream MaxDeletedEntryId mismatch.\nActual: %#v;\nExpected: %#v", info.MaxDeletedEntryId, "0-0")
 			}
 		})
 	}
 }
 
-func TestGetStreamInfoUsingRedis7(t *testing.T) {
+func TestStreamsGetStreamInfoUsingRedis7(t *testing.T) {
 	if os.Getenv("TEST_REDIS7_URI") == "" {
 		t.Skipf("TEST_REDIS7_URI not set - skipping")
 	}
@@ -134,47 +141,46 @@ func TestGetStreamInfoUsingRedis7(t *testing.T) {
 			}
 
 			if info.Length != tst.streamInfo.Length {
-				t.Errorf("Stream length mismatch.\nActual: %#v;\nExpected: %#v\n", info.Length, tst.streamInfo.Length)
+				t.Errorf("Stream length mismatch.\nActual: %#v;\nExpected: %#v", info.Length, tst.streamInfo.Length)
 			}
 			if info.RadixTreeKeys != tst.streamInfo.RadixTreeKeys {
-				t.Errorf("Stream RadixTreeKeys mismatch.\nActual: %#v;\nExpected: %#v\n", info.RadixTreeKeys, tst.streamInfo.RadixTreeKeys)
+				t.Errorf("Stream RadixTreeKeys mismatch.\nActual: %#v;\nExpected: %#v", info.RadixTreeKeys, tst.streamInfo.RadixTreeKeys)
 			}
 			if info.RadixTreeNodes != tst.streamInfo.RadixTreeNodes {
-				t.Errorf("Stream RadixTreeNodes mismatch.\nActual: %#v;\nExpected: %#v\n", info.RadixTreeNodes, tst.streamInfo.RadixTreeNodes)
+				t.Errorf("Stream RadixTreeNodes mismatch.\nActual: %#v;\nExpected: %#v", info.RadixTreeNodes, tst.streamInfo.RadixTreeNodes)
 			}
 			if info.Groups != tst.streamInfo.Groups {
-				t.Errorf("Stream Groups mismatch.\nActual: %#v;\nExpected: %#v\n", info.Groups, tst.streamInfo.Groups)
+				t.Errorf("Stream Groups mismatch.\nActual: %#v;\nExpected: %#v", info.Groups, tst.streamInfo.Groups)
 			}
 			if isNotTestTimestamp(info.LastGeneratedId) {
-				t.Errorf("Stream LastGeneratedId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.LastGeneratedId, TestStreamTimestamps)
+				t.Errorf("Stream LastGeneratedId mismatch.\nActual: %#v;\nExpected any of: %#v", info.LastGeneratedId, TestStreamTimestamps)
 			}
 			if info.FirstEntryId != TestStreamTimestamps[0] {
-				t.Errorf("Stream FirstEntryId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.FirstEntryId, TestStreamTimestamps)
+				t.Errorf("Stream FirstEntryId mismatch.\nActual: %#v;\nExpected any of: %#v", info.FirstEntryId, TestStreamTimestamps)
 			}
 			if info.LastEntryId != TestStreamTimestamps[len(TestStreamTimestamps)-1] {
-				t.Errorf("Stream LastEntryId mismatch.\nActual: %#v;\nExpected any of: %#v\n", info.LastEntryId, TestStreamTimestamps)
+				t.Errorf("Stream LastEntryId mismatch.\nActual: %#v;\nExpected any of: %#v", info.LastEntryId, TestStreamTimestamps)
 			}
 			if info.MaxDeletedEntryId != "0-0" {
-				t.Errorf("Stream MaxDeletedEntryId mismatch.\nActual: %#v;\nExpected: %#v\n", info.MaxDeletedEntryId, "0-0")
+				t.Errorf("Stream MaxDeletedEntryId mismatch.\nActual: %#v;\nExpected: %#v", info.MaxDeletedEntryId, "0-0")
 			}
 		})
 	}
 }
 
-func TestScanStreamGroups(t *testing.T) {
+func TestStreamsScanStreamGroups123(t *testing.T) {
 	if os.Getenv("TEST_REDIS_URI") == "" {
 		t.Skipf("TEST_REDIS_URI not set - skipping")
 	}
 	addr := os.Getenv("TEST_REDIS_URI")
-	db := dbNumStr
 
 	c, err := redis.DialURL(addr)
 	if err != nil {
 		t.Fatalf("Couldn't connect to %#v: %#v", addr, err)
 	}
 
-	if _, err = c.Do("SELECT", db); err != nil {
-		t.Errorf("Couldn't select database %#v", db)
+	if _, err = c.Do("SELECT", dbNumStr); err != nil {
+		t.Errorf("Couldn't select database %#v", dbNumStr)
 	}
 
 	fixtures := []keyFixture{
@@ -185,12 +191,14 @@ func TestScanStreamGroups(t *testing.T) {
 	c.Do("XGROUP", "CREATE", "test_stream_1", "test_group_1", "$", "MKSTREAM")
 	c.Do("XGROUP", "CREATE", "test_stream_2", "test_group_1", "$", "MKSTREAM")
 	c.Do("XGROUP", "CREATE", "test_stream_2", "test_group_2", "$")
+
 	// Add simple values
-	createKeyFixtures(t, c, fixtures)
 	defer func() {
 		deleteKeyFixtures(t, c, fixtures)
 		c.Close()
 	}()
+	createKeyFixtures(t, c, fixtures)
+
 	// Process messages to assign Consumers to their groups
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "test_stream_1", ">")
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "test_stream_2", ">")
@@ -240,38 +248,39 @@ func TestScanStreamGroups(t *testing.T) {
 		t.Run(tst.name, func(t *testing.T) {
 			scannedGroup, _ := scanStreamGroups(c, tst.stream)
 			if err != nil {
-				t.Errorf("Err: %s", err)
+				t.Fatalf("Err: %s", err)
+				return
 			}
 
 			if len(scannedGroup) == len(tst.groups) {
 				for i := range scannedGroup {
 					if scannedGroup[i].Name != tst.groups[i].Name {
-						t.Errorf("Group name mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Name, scannedGroup[i].Name)
+						t.Errorf("%d) Group name mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].Name, scannedGroup[i].Name)
 					}
 					if scannedGroup[i].Consumers != tst.groups[i].Consumers {
-						t.Errorf("Consumers count mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Consumers, scannedGroup[i].Consumers)
+						t.Errorf("%d) Consumers count mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].Consumers, scannedGroup[i].Consumers)
 					}
 					if scannedGroup[i].Pending != tst.groups[i].Pending {
-						t.Errorf("Pending items mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Pending, scannedGroup[i].Pending)
+						t.Errorf("%d) Pending items mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].Pending, scannedGroup[i].Pending)
 					}
 					if parseStreamItemId(scannedGroup[i].LastDeliveredId) != parseStreamItemId(tst.groups[i].LastDeliveredId) {
-						t.Errorf("LastDeliveredId items mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].LastDeliveredId, scannedGroup[i].LastDeliveredId)
+						t.Errorf("%d) LastDeliveredId items mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].LastDeliveredId, scannedGroup[i].LastDeliveredId)
 					}
 					if scannedGroup[i].Lag != tst.groups[i].Lag {
-						t.Errorf("Lag mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Lag, scannedGroup[i].Lag)
+						t.Errorf("%d) Lag mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].Lag, scannedGroup[i].Lag)
 					}
 					if scannedGroup[i].EntriesRead != tst.groups[i].EntriesRead {
-						t.Errorf("EntriesRead mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].EntriesRead, scannedGroup[i].EntriesRead)
+						t.Errorf("%d) EntriesRead mismatch.\nExpected: %#v;\nActual: %#v", i, tst.groups[i].EntriesRead, scannedGroup[i].EntriesRead)
 					}
 				}
 			} else {
-				t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d\n", len(tst.consumers), len(scannedGroup))
+				t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d", len(tst.consumers), len(scannedGroup))
 			}
 		})
 	}
 }
 
-func TestScanStreamGroupsUsingRedis7(t *testing.T) {
+func TestStreamsScanStreamGroupsUsingRedis7(t *testing.T) {
 	if os.Getenv("TEST_REDIS7_URI") == "" {
 		t.Skipf("TEST_REDIS7_URI not set - skipping")
 	}
@@ -291,16 +300,19 @@ func TestScanStreamGroupsUsingRedis7(t *testing.T) {
 		{"XADD", "test_stream_1", []interface{}{"1638006862521-0", "field_1", "str_1"}},
 		{"XADD", "test_stream_2", []interface{}{"1638006862522-0", "field_pattern_1", "str_pattern_1"}},
 	}
+
 	// Create test streams
 	c.Do("XGROUP", "CREATE", "test_stream_1", "test_group_1", "$", "MKSTREAM")
 	c.Do("XGROUP", "CREATE", "test_stream_2", "test_group_1", "$", "MKSTREAM")
 	c.Do("XGROUP", "CREATE", "test_stream_2", "test_group_2", "$")
+
 	// Add simple values
-	createKeyFixtures(t, c, fixtures)
 	defer func() {
 		deleteKeyFixtures(t, c, fixtures)
 		c.Close()
 	}()
+	createKeyFixtures(t, c, fixtures)
+
 	// Process messages to assign Consumers to their groups
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "test_stream_1", ">")
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "test_stream_2", ">")
@@ -356,32 +368,32 @@ func TestScanStreamGroupsUsingRedis7(t *testing.T) {
 			if len(scannedGroup) == len(tst.groups) {
 				for i := range scannedGroup {
 					if scannedGroup[i].Name != tst.groups[i].Name {
-						t.Errorf("Group name mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Name, scannedGroup[i].Name)
+						t.Errorf("Group name mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].Name, scannedGroup[i].Name)
 					}
 					if scannedGroup[i].Consumers != tst.groups[i].Consumers {
-						t.Errorf("Consumers count mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Consumers, scannedGroup[i].Consumers)
+						t.Errorf("Consumers count mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].Consumers, scannedGroup[i].Consumers)
 					}
 					if scannedGroup[i].Pending != tst.groups[i].Pending {
-						t.Errorf("Pending items mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Pending, scannedGroup[i].Pending)
+						t.Errorf("Pending items mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].Pending, scannedGroup[i].Pending)
 					}
 					if parseStreamItemId(scannedGroup[i].LastDeliveredId) != parseStreamItemId(tst.groups[i].LastDeliveredId) {
-						t.Errorf("LastDeliveredId items mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].LastDeliveredId, scannedGroup[i].LastDeliveredId)
+						t.Errorf("LastDeliveredId items mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].LastDeliveredId, scannedGroup[i].LastDeliveredId)
 					}
 					if scannedGroup[i].Lag != tst.groups[i].Lag {
-						t.Errorf("Lag mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].Lag, scannedGroup[i].Lag)
+						t.Errorf("Lag mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].Lag, scannedGroup[i].Lag)
 					}
 					if scannedGroup[i].EntriesRead != tst.groups[i].EntriesRead {
-						t.Errorf("EntriesRead mismatch.\nExpected: %#v;\nActual: %#v\n", tst.groups[i].EntriesRead, scannedGroup[i].EntriesRead)
+						t.Errorf("EntriesRead mismatch.\nExpected: %#v;\nActual: %#v", tst.groups[i].EntriesRead, scannedGroup[i].EntriesRead)
 					}
 				}
 			} else {
-				t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d\n", len(tst.consumers), len(scannedGroup))
+				t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d", len(tst.consumers), len(scannedGroup))
 			}
 		})
 	}
 }
 
-func TestScanStreamGroupsConsumers(t *testing.T) {
+func TestStreamsScanStreamGroupsConsumers(t *testing.T) {
 	if os.Getenv("TEST_REDIS_URI") == "" {
 		t.Skipf("TEST_REDIS_URI not set - skipping")
 	}
@@ -401,15 +413,18 @@ func TestScanStreamGroupsConsumers(t *testing.T) {
 		{"XADD", "single_consumer_stream", []interface{}{"*", "field_1", "str_1"}},
 		{"XADD", "multiple_consumer_stream", []interface{}{"*", "field_pattern_1", "str_pattern_1"}},
 	}
+
 	// Create test streams
 	c.Do("XGROUP", "CREATE", "single_consumer_stream", "test_group_1", "$", "MKSTREAM")
 	c.Do("XGROUP", "CREATE", "multiple_consumer_stream", "test_group_1", "$", "MKSTREAM")
+
 	// Add simple test items to streams
-	createKeyFixtures(t, c, fixtures)
 	defer func() {
 		deleteKeyFixtures(t, c, fixtures)
 		c.Close()
 	}()
+	createKeyFixtures(t, c, fixtures)
+
 	// Process messages to assign Consumers to their groups
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "single_consumer_stream", ">")
 	c.Do("XREADGROUP", "GROUP", "test_group_1", "test_consumer_1", "COUNT", "1", "STREAMS", "multiple_consumer_stream", ">")
@@ -456,15 +471,15 @@ func TestScanStreamGroupsConsumers(t *testing.T) {
 				if len(g.StreamGroupConsumersInfo) == len(tst.consumers) {
 					for i := range g.StreamGroupConsumersInfo {
 						if g.StreamGroupConsumersInfo[i].Name != tst.consumers[i].Name {
-							t.Errorf("Consumer name mismatch.\nExpected: %#v;\nActual: %#v\n", tst.consumers[i].Name, g.StreamGroupConsumersInfo[i].Name)
+							t.Errorf("Consumer name mismatch.\nExpected: %#v;\nActual: %#v", tst.consumers[i].Name, g.StreamGroupConsumersInfo[i].Name)
 						}
 						if g.StreamGroupConsumersInfo[i].Pending != tst.consumers[i].Pending {
-							t.Errorf("Pending items mismatch for %s.\nExpected: %#v;\nActual: %#v\n", g.StreamGroupConsumersInfo[i].Name, tst.consumers[i].Pending, g.StreamGroupConsumersInfo[i].Pending)
+							t.Errorf("Pending items mismatch for %s.\nExpected: %#v;\nActual: %#v", g.StreamGroupConsumersInfo[i].Name, tst.consumers[i].Pending, g.StreamGroupConsumersInfo[i].Pending)
 						}
 
 					}
 				} else {
-					t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d\n", len(tst.consumers), len(g.StreamGroupConsumersInfo))
+					t.Errorf("Consumers entries mismatch.\nExpected: %d;\nActual: %d", len(tst.consumers), len(g.StreamGroupConsumersInfo))
 				}
 			}
 
@@ -472,7 +487,7 @@ func TestScanStreamGroupsConsumers(t *testing.T) {
 	}
 }
 
-func TestExtractStreamMetrics(t *testing.T) {
+func TestStreamsExtractStreamMetrics(t *testing.T) {
 	if os.Getenv("TEST_REDIS_URI") == "" {
 		t.Skipf("TEST_REDIS_URI not set - skipping")
 	}

